### PR TITLE
Improvements to the introductory vignette

### PR DESCRIPTION
**Pull request by Anirban166**
'Improvements' mostly in terms of the sentence structure used, like rephrasing slightly confusing statements or ones with incorrect ways of wording.

Other than that or on the minor end of changes, I fixed a few grammatical errors, avoided the use of `code`s (looks weird, but I guess that could be just me so need more opinions on this!), did one or two addons (like the Oxford comma) for using the British version of English, and fixed minor inconsistencies.

Question for the maintainers and reviewers: 
Assuming such changes would help and I could go about doing the same for the other vignettes (then only), should I make a separate PR for each vignette or would you guys prefer me to commit changes to other vignettes in this PR itself?

Aside: I also see the potential to include hyperlinks to the other vignettes for multiple areas in this document, so if #5833 works then I can include them easily (and I agree that referring to vignettes by their titles would be a good way)

### CODE REVIEW

File: vignettes/datatable-intro.Rmd [hunk 1]
```diff
@@ -479,7 +477,7 @@ head(ans, 10)
 
 #### -- Can `by` accept *expressions* as well or does it just take columns?
 
-Yes it does. As an example, if we would like to find out how many flights started late but arrived early (or on time), started and arrived late etc...
+Yes it does. As an example, if we would like to find out how many flights departed and arrived late, we can do:
```
Comment (HughParsonage) [inline vignettes/datatable-intro.Rmd:480]: Original was correct.

**Review by HughParsonage — APPROVED** [review]
Very minor amendment requested (but not required). This is an improvement thank you.

File: vignettes/datatable-intro.Rmd [hunk 2]
```diff
@@ -564,7 +562,7 @@ head(ans)
 
 ### g) Why keep `j` so flexible?
 
-So that we have a consistent syntax and keep using already existing (and familiar) base functions instead of learning new functions. To illustrate, let us use the `data.table` `DT` that we created at the very beginning under [What is a data.table?](#what-is-datatable-1a) section.
+So that we have a consistent syntax and keep using already existing (and familiar) base functions instead of learning new functions. To illustrate, let us use the `data.table` `DT` that we created at the very beginning under the [What is a data.table?](#what-is-datatable-1a) section.
```
Comment (HughParsonage) [inline vignettes/datatable-intro.Rmd:565]: I'd prefer 'under the section [What is a data.table?]'

**Comment by tdhock** [comment]
looks good to me, thanks for the contribution.

**Comment by MichaelChirico** [comment]
Thank you!

**Pull request merged**